hydromt/region/region.py

```diff
@@ -0,0 +1,85 @@
+"""A class to handle all functionalities to do with model regions."""
+
+from logging import Logger, getLogger
+from os.path import exists
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import numpy as np
+from geopandas import GeoDataFrame, GeoSeries
+
+from hydromt._typing.model_mode import ModelMode
+from hydromt._typing.type_def import StrPath
+from hydromt.region._specifyers import RegionSpecifyer
+
+logger = getLogger(__name__)
+
+
+class Region:
+    """A class to handle all operations to do with the model region."""
+
+    def __init__(
+        self,
+        region_dict: Dict[str, Any],
+        logger: Logger = logger,
+    ):
+        self._dict: Dict[str, Any] = region_dict
+        self._data: Optional[GeoDataFrame] = None
```

Comment:
The geopandas object is the main property of region for the model so I would make it easier to access it.
For example, a ``data`` property that returns ``_data`` or try and calls ``construct`` if it is None?

Might be nice to have shortcut properties to ``crs`` and ``bounds`` (total_bounds) as well to make it easier to use in model but this is really small shortcuts so maybe not worth it (model.region.crs rather than model.region.data.crs)?

Reply:
sure, I've added them.